### ISSUE TITLE
pretrained model and dataset

### ISSUE THREAD

**Issue by zimuyuan** [kaiwang960112/Challenge-condition-FER-dataset#5]
作者你好，最近看了你的论文，想复现你的实验，但没有预训练模型和数据集，你的预训练模型和数据集哪里可以下载呢？没有看到你说的百度网盘链接呢？

**Comment by kaiwang960112** [comment]
最近机房在搞集群，服务器目前都断了，自己也在搞文章，我们在人脸数据集上预训练的模型我会尽快公布出来的，可以拿那个来复现实验。

**Comment by zimuyuan** [comment]
好的，那我会持续关注你们的，希望到时能复现你们的实验！！！

**Comment by kaiwang960112** [comment]
已经把ijba上人脸预训练的模型和vggface训到了89.16的模型放在项目的链接上（百度网盘地址是：https://pan.baidu.com/s/1aXUtIqgGFd1HjkUXhZWqXA 提取码是：fhrm，你们可以下载，按照文章中说的方法去复现，应该问题不大。
数据集ferplus是公开的，邓伟红老师的rafdb和丹佛大学的affectnet都是需要申请的，我没有开源权限，你们可以去申请，一周可以下来。

**Comment by zimuyuan** [comment]
作者你好，我现在下载了ferplus数据集，准备运行你FERplus_dir这个目录的代码，可是我发现里面缺少了几个py文件，attention_rank_loss.py ，test_rank_loss_attention.py ，val_part_attention_sample.py 这些在下载包里面都没能看见。请问这些是否也是可以开源的呢，或者可以分享给个人吗？因为这是我第一次尝试复现论文实验，所以有很多问题可能不太懂。希望作者能理解。如果作者觉得方便是否可以加一下QQ或者微信好友。总之还是非常感谢作者所做的工作！！！

**Comment by shehzi-khan** [comment]
I am unable to extract this file https://drive.google.com/drive/folders/1dpDntw-2jLCNgjZ20W9aLg8a6DEEo7gX

**Comment by kaiwang960112** [comment]
I click the link, it has no problem. Could you try again?

**Comment by kaiwang960112** [comment]
我整理下，尽快开源。谢谢对我们工作的关注

**Comment by kaiwang960112** [comment]
你先跑那个不带rank版本的。

**Comment by zimuyuan** [comment]
没事，那可以先等你把这个整理好先，因为AffectNet这个数据集实在太大了，我还没能下载下来，即使下载了可能也很难跑起来。所以我现在只是尝试在FER+这个数据集上训练测试。不过你的下载链接下解压出来的FERplus_dir这个目录里面只有part_attention.py    part_attention_sample.py    train_attention_rank_loss.py 这三个py文件似乎跑步起来。

**Comment by kaiwang960112** [comment]
可能路径什么的你有点儿问题吧。ferplus数据集你下载了吗？

**Comment by shehzi-khan** [comment]
The link is working, but after downloading the file (approx 256.96 MB) when I try to untar or uncompress the file, I get error "File is not an archive". I tried on Ubuntu using tar command as well as Windows using WinRar.

**Comment by zimuyuan** [comment]
@kaiwang960112 我只直接在GitHub下载到Win10上的，然后直接解压出来的，路径上应该没有什么问题的。只是感觉py文件太少了，这里不能截图，不然可以截图给你看看的。FERplus_dir这个目录里面只有part_attention.py part_attention_sample.py train_attention_rank_loss.py  这有这三个文件和我的FER+数据集能跑起来吗？感觉好像还不行呢

**Comment by kaiwang960112** [comment]
part_attention.py 是模型定义文件，part_attention_sample.py是构dataset的代码， train_attention_rank_loss.py 是训练代码，至于文件路径的txt你可以自己去生成。因为我觉得每个人电脑里面的路径不一样。可能你需要花点时间把路径和txt做好。加油

**Comment by kaiwang960112** [comment]
maybe you can download the model from baidu cloud link.

**Comment by zimuyuan** [comment]
@kaiwang960112 作者你好，最近在看你的代码，看到这里有些不太明白，part_attention.py这个文件里面的ResNet(nn.Module)这个里面的最后一层self.fc = nn.Linear(512,8)输出是8维的，可是forward(self, x)函数里面返回的self_pred_score = self.fc(vm)。按理来说也是8维的吧。可是train_attention_rank_loss.py文件里的train()函数
pred_score, alphas_part_max, alphas_org = model(input_var)
模型怎么会有三个返回，另两个alphas_part_max，alphas_org，不懂从哪里来。我基础比较差！！

**Comment by shehzi-khan** [comment]
attention_rank_loss.py
test_rank_loss_attention.py
val_part_attention_sample.py

**Comment by shehzi-khan** [comment]
ijba_res18_naive.pth.tar: The archive is either in unknown format or damaged
No archives found

**Comment by shehzi-khan** [comment]
please tell me how to use this pre_trained model

**Comment by jtheiner** [comment]
The authors have only provided the pytorch model and no inference script. Hence, you have to write it yourself using 'torch.load("ijba_res18_naive.pth.tar")'.

**Comment by shehzi-khan** [comment]
Thanks @jtheiner 
Any suggestion about missing files?

1. attention_rank_loss.py
2. test_rank_loss_attention.py
3. val_part_attention_sample.py

**Comment by kaiwang960112** [comment]
I have uploaded the 'attention_rank_loss.py', 'test_rank_loss_attention.py' and 'val_part_attention_sample.py'. I am very sorry. @shehzi-khan 
I offer the train code, in 142 line, you can find how to use the ijba_res18_naive.pth.tar.

**Comment by kaiwang960112** [comment]
Thank you very much!!!@jtheiner

**Comment by zimuyuan** [comment]
@kaiwang960112 作者你好，我试着运行了你的train_attention_rank_loss.py这个文件来训练模型，可是我把训练batch改成了64，可是发现占用CPU很多，我看你的代码里面也用了.cuda()加载到GPU计算了啊。请问这是什么原因呢？你训练的时候也是这样吗？

**Comment by kaiwang960112** [comment]
我跑的时候没遇到这样的情况，你现在跑的时候cpu占用多，gpu占用也多吗？

**Comment by zimuyuan** [comment]
CPU占用了50%几，GPU占用大概是6800M多这样，下面是输出结果
`Epoch: [0][0/442]	Time 68.11069536209106 (68.11069536209106)	Data 61.351630210876465 (61.351630210876465)	Loss 2.1900947093963623 (2.1900947093963623)	weights_loss 0.09383542835712433 (0.09383542835712433)	Prec@1 23.4375 (23.4375)	
Epoch: [0][10/442]	Time 68.52843427658081 (62.211667190898545)	Data 67.22116565704346 (60.4200664433566)	Loss 1.8463870286941528 (3.0376737117767334)	weights_loss 0.09163790196180344 (0.09287703104994514)	Prec@1 42.1875 (30.397727966308594)	
`而且每个mini-batch(batch size = 64)训练的时间很长，输出的Time 68.11069536209106 (68.11069536209106) ，这样正常吗，怎么感觉训练的那么慢的，估计一个小时也训练不完一个周期，是不是我的没有成功加载到GPU上训练啊

**Comment by kaiwang960112** [comment]
单卡 这个gpu占用应该是正常的。你训得是什么数据集？目测是ferplus？是不是进程数没开够？

**Comment by zimuyuan** [comment]
那CPU占用率和你的一样吗，差不多60%了。是图片的加载过程预处理过程占用CPU比较多吗？嗯数据是FERPLUS。我用的是单进程，所以我的训练比较慢吗。

**Comment by kaiwang960112** [comment]
应该是这个问题 你把进程数加上去看看能不能解决这个问题

**Comment by zimuyuan** [comment]
因为我是在Windows的服务器下训练的，好像加了进程就会报错的，后面就改成单进程了。

**Comment by YileYile** [comment]
@zimuyuan 你好，我看你已经运行出代码了，能否加一下QQ好友请教你一下啊，非常感谢！

**Comment by zimuyuan** [comment]
@YileYile 可以的啊我的qq号:292681029

**Comment by zimuyuan** [comment]
@kaiwang960112 作者你好，你的人脸检测对齐裁剪的代码可以上传吗？我没有找到相应的代码。

**Comment by cimszw** [comment]
@kaiwang960112 你好！祝论文顺利！
我也下载了Google drive(有一个)和百度网盘的.tar(有两个，包括一个checkpoit)文件。
链接正常可以下载
但在windows平台上用winrar(其它的.tar可以用它解压的)无法解压，尝试打开就会报错:未知格式或者压缩文件损坏。
两个问题：
1. 环境信息是：Python 2还是3？
2. 是在linux下还是windows下跑的?

**Comment by XiaoWang98** [comment]
@cimszw 我在linux下python2.7运行起来了

**Comment by cimszw** [comment]
预训练的模型可以解压使用么？

> @cimszw 我在linux下python2.7运行起来了

**Comment by XiaoWang98** [comment]
> 预训练的模型可以解压使用么？
> 
> > @cimszw 我在linux下python2.7运行起来了

直接加载啊

**Comment by QingfengLi-ing** [comment]
> > 预训练的模型可以解压使用么？
> > > @cimszw 我在linux下python2.7运行起来了
> 
> 直接加载啊

您好，我想问一下，我直接加载提示错误，网上的方法并不能解决， 感觉应该是定义模型的代码不一样导致的，想问一下您能不能分享一下vgg16模型的代码
Missing key(s) in state_dict: "features.0.weight", "features.0.bias", "features.1.weight", "features.1.bias", "features.1.running_mean", "features.1.running_var", "features.3.weight", "features.3.bias", "features.4.weight", "features.4.bias", "features.4.running_mean", "features.4.running_var", "features.7.weight", "features.7.bias", "features.8.weight", "features.8.bias", "features.8.running_mean", "features.8.running_var", "features.10.weight", "features.10.bias", "features.11.weight", "features.11.bias", "features.11.running_mean", "features.11.running_var", "features.14.weight", "features.14.bias", "features.15.weight", "features.15.bias", "features.15.running_mean", "features.15.running_var", "features.17.weight", "features.17.bias", "features.18.weight", "features.18.bias", "features.18.running_mean", "features.18.running_var", "features.20.weight", "features.20.bias", "features.21.weight", "features.21.bias", "features.21.running_mean", "features.21.running_var", "features.24.weight", "features.24.bias", "features.25.weight", "features.25.bias", "features.25.running_mean", "features.25.running_var", "features.27.weight", "features.27.bias", "features.28.weight", "features.28.bias", "features.28.running_mean", "features.28.running_var", "features.30.weight", "features.30.bias", "features.31.weight", "features.31.bias", "features.31.running_mean", "features.31.running_var", "features.34.weight", "features.34.bias", "features.35.weight", "features.35.bias", "features.35.running_mean", "features.35.running_var", "features.37.weight", "features.37.bias", "features.38.weight", "features.38.bias", "features.38.running_mean", "features.38.running_var", "features.40.weight", "features.40.bias", "features.41.weight", "features.41.bias", "features.41.running_mean", "features.41.running_var", "classifier.0.weight", "classifier.0.bias", "classifier.3.weight", "classifier.3.bias", "classifier.6.weight", "classifier.6.bias". 
	Unexpected key(s) in state_dict: "loss", "epoch", "optimizer", "best_prec1", "loss_param", "state_dict", "prec1", "arch".

**Comment by LiuYeeJ** [comment]
> > 预训练的模型可以解压使用么？
> > > @cimszw我在linux下python2.7运行起来了
> 
> 直接加载啊

你好，可以加一下微信或者QQ请教一下问题吗，因为我现在还没有运行出来

**Comment by LiuYeeJ** [comment]
> @kaiwang960112 作者你好，我试着运行了你的train_attention_rank_loss.py这个文件来训练模型，可是我把训练batch改成了64，可是发现占用CPU很多，我看你的代码里面也用了.cuda()加载到GPU计算了啊。请问这是什么原因呢？你训练的时候也是这样吗？

想请教一下你都进行了哪些更改呢，我总是提示CUDA错误，但是GPU的端口号已经都改了啊。

**Comment by yymaitxm** [comment]
你们的.py文件在哪里下载的呀？
